src/domains/template_generator_base.cpp

```diff
@@ -282,7 +282,8 @@ void template_generator_baset::add_var(
   {
     const array_typet &array_type=to_array_type(var.type());
     mp_integer size;
-    to_integer(array_type.size(), size);
+    if(array_type.size().is_constant())
```

Comment:
I think this would be easier to understand if we just returned here on non-constant-size arrays, i.e. we just skip them.

Reply:
Makes sense. We still have to cast it though due to the API but it is a bit more clear that the for loop is ignored if it is not constant, fixed.